api.py

```diff
@@ -83,26 +85,32 @@ def getNowPlaying(self, sources: Optional[List[Source]] = None, allow_off_air: O
             }}
           }}
         }}
-      '''.format(params)
+      '''.format(param_str)
     )
 
     #try:
-    response: Optional[NowPlaying] = self.client.execute(query)
+    response: Optional[Dict[str,Any]] = self.client.execute(query)
     #except:
     #  raise # TODO: Log this.
     #  return None
 
-    if response != None:
-      if ("nowPlaying" in response and response["nowPlaying"]):
-        start_time: int = int(datetime.datetime.fromisoformat(response["nowPlaying"]["startTime"]).timestamp())
-        track: Track = response["nowPlaying"]["track"]
+    if response != None and "nowPlaying" in response and response["nowPlaying"]:
+      start_time: int = int(datetime.datetime.fromisoformat(response["nowPlaying"]["startTime"]).timestamp())
+      track: Track = response["nowPlaying"]["track"]
 
-        nowPlaying: NowPlaying = {
-          "start_time": start_time,
-          "track": track
-        }
-        return nowPlaying
-    return None
+      # A manual tracklist doesn't have a length,
+      # Just assume an average song length ish.
+      if "length" not in track:
```

Comment:
This bit might still be useful, but I don't think I like the premis of the rest of this MR anymore.